File: backend/src/main/java/com/festago/festival/repository/FestivalRepository.java
package com.festago.festival.repository;

import com.festago.common.exception.ErrorCode;
import com.festago.common.exception.NotFoundException;
import com.festago.festival.domain.Festival;
import java.util.Optional;
import org.springframework.data.repository.Repository;

public interface FestivalRepository extends Repository<Festival, Long>, FestivalRepositoryCustom {

    default Festival getOrThrow(Long festivalId) {
        return findById(festivalId)
            .orElseThrow(() -> new NotFoundException(ErrorCode.FESTIVAL_NOT_FOUND));
    }

    boolean existsBySchoolId(Long schoolId);

    Festival save(Festival festival);

    Optional<Festival> findById(Long festivalId);

    void deleteById(Long festivalId);

    void flush();

    boolean existsById(Long festivalId);
}

File: android/festago/app/src/main/java/com/festago/festago/domain/repository/FestivalRepository.kt
package com.festago.festago.domain.repository

import com.festago.festago.domain.model.Festival
import com.festago.festago.domain.model.Reservation

interface FestivalRepository {
    suspend fun loadFestivals(): Result<List<Festival>>
    suspend fun loadFestivalDetail(festivalId: Long): Result<Reservation>
}
